Add pronoun macros feature for persona management

Samueras opened the pull request:
Adds the Makros 
{{pronoun.subjective}}
{{pronoun.objective}}
{{pronoun.pos_det}}
{{pronoun.pos_pro}}
{{pronoun.reflexive}}
And add Pronoun Settings to the persona management.
Macros are identical to the once used on Wyvern.chat

<!-- Put X in the box below to confirm -->

## Checklist:

- [x ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

Samueras committed:
Add pronoun macros feature

- Add 5 new pronoun macros: {{pronoun.subjective}}, {{pronoun.objective}}, {{pronoun.pos_det}}, {{pronoun.pos_pro}}, {{pronoun.reflexive}}
- Add pronoun settings UI to persona management with 5 input fields
- Add 4 preset buttons: She/Her, He/Him, They/Them, It/Its
- Update persona data structure to include pronoun fields
- Integrate pronoun macros with existing macro system
- Style UI to match existing SillyTavern design

Samueras committed:
Fix pronoun preset button click handler

- Use event.currentTarget instead of event.target to ensure clicks on button text work properly
- This fixes the issue where clicking on the text inside the button didn't trigger the preset

Samueras committed:
Fix ESLint issues in personas.js

- Remove trailing spaces
- Add missing trailing commas in pronoun objects
- Fix code formatting to meet SillyTavern linting standards
- All ESLint errors resolved

Wolfsblvt commented on public/scripts/personas.js:
```diff
@@ -855,6 +882,11 @@ async function selectCurrentPersona({ toastPersonaNameChange = true } = {}) {
                 lorebook: '',
                 connections: [],
                 title: '',
+                pronounSubjective: '',
+                pronounObjective: '',
+                pronounPosDet: '',
+                pronounPosPro: '',
+                pronounReflexive: '',
```
If we are adding new fields, I'd prefer to add them structured.
So `pronoun.subjective`, etc. Similarly to how the macros are named. Works better than several root fields with the same prefix.

Wolfsblvt commented on public/script.js:
```diff
@@ -2283,6 +2283,23 @@ export function substituteParams(content, _name1, _name2, _original, _group, _re
     environment.groupNotMuted = getGroupValue(false);
     environment.model = getGeneratingModel();
 
+    // Add pronoun macros from current persona
+    if (power_user.persona_descriptions && power_user.persona_descriptions[user_avatar]) {
+        const personaData = power_user.persona_descriptions[user_avatar];
+        environment['pronoun.subjective'] = personaData.pronounSubjective || '';
+        environment['pronoun.objective'] = personaData.pronounObjective || '';
+        environment['pronoun.pos_det'] = personaData.pronounPosDet || '';
+        environment['pronoun.pos_pro'] = personaData.pronounPosPro || '';
+        environment['pronoun.reflexive'] = personaData.pronounReflexive || '';
+    } else {
+        // Default empty values if no persona data
+        environment['pronoun.subjective'] = '';
+        environment['pronoun.objective'] = '';
+        environment['pronoun.pos_det'] = '';
+        environment['pronoun.pos_pro'] = '';
+        environment['pronoun.reflexive'] = '';
+    }
```
You could simplify this by making `personaData` optional and null based on the same if check, and then do something like `environment['pronoun.subjective'] = personaData?.pronounSubjective || '';`

Wolfsblvt commented on public/scripts/personas.js:
```diff
@@ -572,6 +582,23 @@ export function setPersonaDescription() {
         .prop('selected', String(true));
     $('#persona_lore_button').toggleClass('world_set', !!power_user.persona_description_lorebook);
+
+    // Load pronoun values from current persona
+    const personaData = power_user.persona_descriptions?.[user_avatar];
+    if (personaData) {
+        $('#persona_pronoun_subjective').val(personaData.pronounSubjective || '');
+        $('#persona_pronoun_objective').val(personaData.pronounObjective || '');
+        $('#persona_pronoun_pos_det').val(personaData.pronounPosDet || '');
+        $('#persona_pronoun_pos_pro').val(personaData.pronounPosPro || '');
+        $('#persona_pronoun_reflexive').val(personaData.pronounReflexive || '');
+    } else {
+        $('#persona_pronoun_subjective').val('');
+        $('#persona_pronoun_objective').val('');
+        $('#persona_pronoun_pos_det').val('');
+        $('#persona_pronoun_pos_pro').val('');
+        $('#persona_pronoun_reflexive').val('');
+    }
```
Same here, no need to do the if/else, you can just make the optional `personaData?.pronounSubjective` access.

Wolfsblvt reviewed (commented):
Generally like the idea. Makes sense.

I have a few suggestions/code improvements.
Have to test it and play around a bit with it too.

Wolfsblvt commented:
/gemini review

Samueras committed:
Refactor pronoun fields to use nested structure

- Change from flat structure (pronounSubjective, etc.) to nested structure (pronoun.subjective, etc.)
- Update all object definitions in personas.js (8 locations)
- Update form reading/writing code to use nested structure
- Update environment/macro code in script.js
- Maintains backward compatibility with optional chaining
- Addresses code review feedback for better structure consistency

Samueras committed:
Simplify redundant if/else blocks in pronoun handling

- Remove redundant else blocks in script.js and personas.js
- Use nullish coalescing (?? {}) for cleaner fallback handling
- Maintain same functionality with fewer lines of code
- Addresses code review feedback about redundant logic

Samueras committed:
Refactor pronoun preset switch statement to object map

- Replace verbose switch statement with clean object map
- Reduce code from ~50 lines to ~15 lines
- Make adding new pronoun presets much easier
- Improve code readability and maintainability

Samueras committed:
Fix ESLint trailing spaces in personas.js

Samueras commented:
@Wolfsblvt I'm really not that good at coding. But have removed the if-elses as Gemini had suggested. 
I am not sure what you mean with 
> You could simplify this by making personaData optional and null based on the same if check, and then do something like environment['pronoun.subjective'] = personaData?.pronounSubjective || '';

I guess it isn't needed anymore? But if it is, please let me know.

Cohee1207 commented:
@Samueras how comfortable would you be to move this functionality to an extension and potentially submit it to the content repo (downloadable assets list)? We feel like expanding the fields of personas or characters with non-use case agnostic fields is better fit for an extension. If you need any guidance or help on how to create the extension or want us to take it over from here, let me know.

Samueras commented:
@Cohee1207 
I am a bit surprised. It felt to me as this would make more sense as a core function than an extension. I know a PoVs for cards is an ongoing issue for many. Having it will ensure compatibility with cards from Wyfern as well. And I wanted to address the issue with lore/data from Chub as well to see if they would include that into Chub once ST would have it included.

So I would really love this to be a core function. And make other sites to implement it as well, but if you prefer it to be an extension, then that is totally fine. 

I would love for you to take over in that case, though.

Wolfsblvt commented:
Personally, I feel like this is a nice idea in concept, but I am with Cohee here. The use-case is quite specific. The last surveys we did showed most users of ST rarely even use more than a handful of personas.

Additional features  and more customizations are always great and appreciated, but we kinda have adopted an "extension first" approach with many of those requests and ideas. Having it as a core extension makes it easy enough to enable for anyone.

Another argument is, that a lot of people already have solutions for this for ages. Simply using macros and variables. Something like this in the persona description:
```
{{setvar::pronoun.objective::he}}
{{setvar::pronoun.objective::him}}
```
And then wherever you want you can just write it like `{{char}} met {{getvar::pronoun.objective}} at an arcade.`.

I can see if I find the time to convert this into an extension.

Wolfsblvt commented:
Hey @Samueras, I took the base idea and implementation you did and wrote an extension on top of it.
It's still in the works and might change a bit, but wanted to let you know.
https://github.com/Wolfsblvt/SillyTavern-Pronouns

Likely will release it soon. But feel free to test it and give me feedback.

Samueras commented:
Thank you so much for doing that and letting me know.

Cohee1207 commented:
I'll close it now since it's moved to a downloadable extension.
